packages/patternfly-4/react-core/src/layouts/Page/Page.js

```diff
@@ -33,54 +34,71 @@ const defaultProps = {
   scrollingDistance: 20
 };
 
-class Page extends React.Component {
-  state = { isCondensed: false };
+export const PageContext = React.createContext();
 
+class Page extends React.Component {
   constructor(props) {
     super(props);
     this.mainRef = React.createRef();
+
+    this.state = {
+      isTall: this.isTall(props)
+    };
   }
   componentDidMount() {
     const { useCondensed } = this.props;
     if (useCondensed) {
       // I picked this because it's approx 1 frame (ie: 16.7ms)
       this.mainRef.current.addEventListener('scroll', debounce(this.handleScroll, 16));
+      window.addEventListener('resize', this.handleResize);
     }
   }
   componentWillUnmount() {
     const { useCondensed } = this.props;
     if (useCondensed) {
       this.mainRef.current.removeEventListener('scroll', this.handleScroll);
+      window.removeEventListener('resize', this.handleResize);
     }
   }
 
+  // only enable tall header if the user is using the `useCondensed` header feature and we are on desktop
+  isTall = props =>
+    typeof window !== 'undefined' && window.innerWidth >= parseInt(breakpointMd.value, 10) && props.useCondensed;
```

Comment:
Nit: Maybe check props.useCondensed first

Reply:
giving this a spin..